apps/application/serializers/application_serializers.py

```diff
@@ -1110,12 +1113,38 @@ def application_list(self, with_valid=True):
                 self.is_valid(raise_exception=True)
             user_id = self.data.get('user_id')
             application_id = self.data.get('application_id')
-            application = Application.objects.filter(user_id=user_id).exclude(id=application_id)
+            application = QuerySet(Application).get(id=application_id)
+
+            application_user_id = user_id if user_id == str(application.user_id) else None
+
+            if application_user_id is not None:
+                all_applications = Application.objects.filter(user_id=application_user_id).exclude(id=application_id)
+            else:
+                all_applications = Application.objects.none()
+
+            # 获取团队共享的应用
+            shared_applications = Application.objects.filter(
+                id__in=TeamMemberPermission.objects.filter(
+                    auth_target_type='APPLICATION',
+                    operate__contains=RawSQL("ARRAY['USE']", []),
+                    member_id__team_id=application.user_id,
+                    member_id__user_id=user_id
+                ).values('target')
+            )
+            all_applications = all_applications.union(shared_applications)
+
             # 把应用的type为WORK_FLOW的应用放到最上面 然后再按名称排序
-            serialized_data = ApplicationSerializerModel(application, many=True).data
+            serialized_data = ApplicationSerializerModel(all_applications, many=True).data
             application = sorted(serialized_data, key=lambda x: (x['type'] != 'WORK_FLOW', x['name']))
             return list(application)
 
+        def get_application(self, app_id, with_valid=True):
+            if with_valid:
+                self.is_valid(raise_exception=True)
+            application = QuerySet(Application).filter(id=self.data.get("application_id")).first()
+            return ApplicationSerializer.Operate(data={'user_id': application.user_id, 'application_id': app_id}).one(
+                with_valid=True)
+
     class ApplicationKeySerializerModel(serializers.ModelSerializer):
         class Meta:
             model = ApplicationApiKey
```

Comment:
### Code Review and Suggestions

#### General Improvements:

1. **Imports**: Ensure imports are organized and clear, particularly around the `RawSQL` usage.
2. **Code Readability**: Simplify code structure where possible to improve readability.

#### Specific Issues/Improvements:

1. **`application_list` Method**:
   - The logic for handling multiple conditions related to team permissions might be complex. Consider simplifying this logic.
   - Use of raw SQL should be kept minimal unless absolutely necessary. If you need dynamic filtering based on roles/access levels, consider using Django's ORM capabilities for more maintainable code.

2. **`get_application` Method**:
   - Similar to `application_list`, the logic for sharing applications seems overly complex. Use a single query to fetch required data without duplicating logic.

#### Optimizations:

1. **Avoid Redundant Queries**:
   - Instead of querying twice (`Application` objects first, then union with other filters), streamline the process with a single query that considers both cases.

2. **Use Django ORM Features**:
   - Explore whether Django's ORM features can simplify your queries and make them more readable.

Here is an optimized version of the `application_list` method considering some of these improvements:

```python
def application_list(self, with_valid=True):
    if with_valid:
        self.is_valid(raise_exception=True)

    user_id = self.data.get('user_id')
    application_id = self.data.get('application_id')

    application = Application.objects.filter(id=application_id).first()

    if application and user_id == str(application.user_id):
        return list(sorted([
            dict(item)
            for item in ApplicationSerializerModel(
                Application.objects.all(), many=True).data
            if item['id'] != application_id
            and item['type'] != 'WORK_FLOW'
        ], key=lambda x: (not x.get('type', '') == 'WORK_FLOW', x.get('name'))))
    elif application or AuthOperate.objects.filter(name="USE").exists():
        shared_apps_qs = TeamMemberPermission.objects.filter(
            auth_target_type='APPLICATION',
            operate__contains=[AuthOperate.objects.first()],
            member_id__team_id=application.user_id,
            member_id__user_id=user_id
        ).values_list('target_id', flat=True)
        
        all_applications = (Application.objects.exclude(id=application_id) |
                            Application.objects.filter(id__in=set(shared_apps_qs)))

        return list(sorted(all_applications.annotate(display_name=Val(application.name)) |
                          Application.objects.filter(type='WORK_FLOW'),
                         key=lambda x: (True if x.type != 'WORK_FLOW' else False, x.display_name)))
    
    return []
```

This revision reduces redundancy by combining multiple queries into one and simplifies condition checks for teams and individual users. Note that the use of `Val` and `annotate` here assumes you have access to the correct methods/functions to achieve similar functionality. Adjustments to match actual Django versions or project configurations may be needed.